lib/qa/authorities/discogs.rb

```diff
@@ -0,0 +1,130 @@
+require 'rdf'
+module Qa::Authorities
+  class Discogs < Base
+    extend ActiveSupport::Autoload
+    autoload :DiscogsTranslation
+    autoload :DiscogsUtils
+    autoload :DiscogsWorksBuilder
+    autoload :DiscogsInstanceBuilder
+    include WebServiceBase
+    include DiscogsTranslation
+
+    class_attribute :discogs_secret, :discogs_key
+    attr_accessor :primary_artists
+
+    def initialize
+      self.primary_artists = []
+    end
+
+    def search(q, tc)
+      # Discogs distinguishes between masters and releases, where a release represents a specific
+      # physical or digital object and a master represents a set of similar releases. Use of a
+      # subauthority (e.g., /qa/search/discogs/master) will target a specific type. No subauthority
+      # will search for both types.
+      unless discogs_key && discogs_secret
+        Rails.logger.error "Questioning Authority tried to call Discogs, but no secret and/or key were set."
+        return []
+      end
+      subauthority = tc.params["subauthority"].present? ? tc.params["subauthority"] : ""
+      parse_authority_response(json(build_query_url(q, subauthority)))
+    end
+
+    def find(id, tc)
+      # If the requested format is json-ld, call the build_graph method in the translation module;
+      # otherwise, just return the Discogs json. If there's no subauthority, check the response
+      # to determine if it should go to the translation module.
+      response = tc.params["subauthority"].present? ? json(find_url(id, tc.params["subauthority"])) : fetch_discogs_results(id)
+      return build_graph(response) unless tc.params["format"] != "jsonld" || response["message"].present?
+      response
+    end
+
+    def build_query_url(q, subauthority)
+      escaped_q = URI.escape(q)
```

Comment:
The use of `URI.escape` is deprecated.  A recent [PR](https://github.com/samvera/questioning_authority/pull/210) switched to using `ERB::Util.url_encode`.

Reply:
done